src/commands/basemaps-github/create-mapsheet.ts

```diff
@@ -0,0 +1,106 @@
+import { ConfigBundled, ConfigProviderMemory } from '@basemaps/config';
+import { Bounds } from '@basemaps/geo';
+import { fsa } from '@chunkd/fs';
+import { command, option, optional, string } from 'cmd-ts';
+import * as fgb from 'flatgeobuf/lib/mjs/geojson.js';
+import { FeatureCollection, MultiPolygon } from 'geojson';
+
+import { CliInfo } from '../../cli.info.js';
+import { logger } from '../../log.js';
+import { registerCli, verbose } from '../common.js';
+
+interface Output {
+  sheetCode: string;
+  files: string[];
+}
+
+export const CommandCreateMapSheetArgs = {
+  verbose,
+  path: option({
+    type: string,
+    long: 'path',
+    description: 'Path of flatgeobuf, this can be both a local path or s3 location',
+  }),
+  config: option({
+    type: string,
+    long: 'config',
+    description: 'Path of basemaps config json, this can be both a local path or s3 location',
+  }),
+  output: option({
+    type: string,
+    long: 'output',
+    description: 'Output of the mapsheet file',
+  }),
+  include: option({
+    type: optional(string),
+    long: 'include',
+    description: 'Include the layers with the pattern in the layer name.',
+  }),
+  exclude: option({
+    type: optional(string),
+    long: 'exclude',
+    description: 'Exclude the layers with the pattern in the layer name.',
+  }),
+};
+
+export const basemapsCreateMapSheet = command({
+  name: 'bm-create-mapsheet',
+  version: CliInfo.version,
+  description: 'Create a cog mapsheet from provided flatgeobuf',
+  args: CommandCreateMapSheetArgs,
+  async handler(args) {
+    registerCli(this, args);
+    const path = args.path;
+    const config = args.config;
+    const outputPath = args.output;
+
+    const include = args.include ? new RegExp(args.include.toLowerCase(), 'i') : undefined;
+    const exclude = args.exclude ? new RegExp(args.exclude.toLowerCase(), 'i') : undefined;
+
+    logger.info({ path }, 'MapSheet:LoadFgb');
+    const buf = await fsa.read(path);
+    logger.info({ config }, 'MapSheet:LoadConfig');
+    const configJson = await fsa.readJson<ConfigBundled>(config);
+    const men = ConfigProviderMemory.fromJson(configJson);
+
+    const rest = fgb.deserialize(buf) as FeatureCollection;
+
+    const aerial = await men.TileSet.get('ts_aerial');
+    if (aerial == null) throw new Error('Invalid config file.');
+    const layers = aerial.layers.filter(
+      (c) => c[2193] != null && (c.maxZoom == null || c.maxZoom > 19) && (c.minZoom == null || c.minZoom < 32),
+    );
+    const imageryIds = new Set<string>();
+    for (const layer of layers) {
+      if (exclude && exclude.test(layer.name)) continue;
+      if (include && !include.test(layer.name)) continue;
+      if (layer[2193] != null) imageryIds.add(layer[2193]);
+    }
+    const imagery = await men.Imagery.getAll(imageryIds);
+
+    const output: Output[] = [];
```

Comment:
An array of `Output` entries should probably have a plural name, such as `outputs`.